services/learn-card-network/brain-service/test/profiles.spec.ts

```diff
@@ -1698,4 +1492,112 @@ describe('Profiles', () => {
             });
         });
     });
-});
+
+    describe('Invite-related tests', () => {
+        beforeEach(async () => {
+            await Profile.delete({ detach: true, where: {} });
+            await userA.clients.fullAuth.profile.createProfile({ profileId: 'usera' });
+            await userB.clients.fullAuth.profile.createProfile({ profileId: 'userb' });
+            await userC.clients.fullAuth.profile.createProfile({ profileId: 'userc' });
+        });
+
+        describe('generateInvite', () => {
+            it('creates a new UUID challenge if none is supplied', async () => {
+                const result = await userA.clients.fullAuth.profile.generateInvite();
+                expect(result.challenge).toBeTruthy();
+                expect(result.challenge).toMatch(
+                    /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/
+                );
+            });
```

Comment:
```suggestion
            it('creates a new challenge if none is supplied', async () => {
                const result = await userA.clients.fullAuth.profile.generateInvite();
                expect(result.challenge).toBeTruthy();
            });
```

This doesn't necessarily have to be a UUID!